app/models/queen.rb

```diff
@@ -1,5 +1,18 @@
-class Queen < Piece
-  scope :piece, -> {where(type: 'Queen')}
-
-
+class Queen < Piece
+  scope :piece, -> {where(type: 'Queen')}
+  
+  def valid_move?(position_x, position_y)
+    Rook.new(position_x,position_y).valid_move? or Bishop.new(position_x, position_y)

```

Comment:
@gvin10 if you want to call valid_move? on the newly created Rook, you need to pass it the coordinates x and y.. I like how you leveraged the code of rook and bishop here.. a bit cleaner way would be to abstract out the implementation of rook and bishop so that the queen does not directly depend on the other pieces, but that they all use the abstracted logic.. i.e. having a `valid_straight?` and `valid_diagonal?`methods on a piece and use them for the rook and bishop respectively and both of them on the queen.. just as a direction.. hope it makes sense